syntax = "proto3";
import "optimizers.proto";

message Layer {
	enum Type {
	     DENSE = 0;
	}
	int32 input_dimension = 1;
	int32 output_dimension = 2;
}

message Epoch {
	float num_seconds = 1;
	float loss = 2;
	float train_accuracy = 3;
}

message Experiment {
	string start_date_time = 1; // from str(datetime.datetime.now())
	string end_date_time = 2; //   in python

	repeated Layer structure = 3;
	repeated Epoch results = 4;
	oneof optimizer {
	      SGD sgd = 5;
	      RMSprop rmsprop = 6;
	      Adagrad adagrad = 7;
	      Adadelta adadelta = 8;
	      Adam adam = 9;
	      Adamax adamax = 10;
	}
	enum Dataset {
	     OSU_TTBAR = 0;
	     OSU_TTHIGGS = 1;
	}
	string description = 11;
}